src/layout/Summary2/config.tsx

```diff
@@ -0,0 +1,67 @@
+import { CG, Variant } from 'src/codegen/CG';
+import { CHECKBOX_SUMMARY_PROPS } from 'src/layout/Checkboxes/config';
+import { CompCategory } from 'src/layout/common';
+import { DROPDOWN_SUMMARY_PROPS } from 'src/layout/Dropdown/config';
+import { INPUT_SUMMARY_PROPS } from 'src/layout/Input/config';
+import { MULTIPLE_SELECT_SUMMARY_PROPS } from 'src/layout/MultipleSelect/config';
+import { RADIO_SUMMARY_PROPS } from 'src/layout/RadioButtons/config';
+export const Config = new CG.component({
+  category: CompCategory.Container,
+  rendersWithLabel: false,
+  capabilities: {
+    renderInTable: false,
+    renderInButtonGroup: false,
+    renderInAccordion: false,
+    renderInAccordionGroup: false,
+    renderInCards: false,
+    renderInCardsMedia: false,
+  },
+})
+  .addProperty(
+    new CG.prop(
+      'children',
+      new CG.arr(new CG.str())
+        .optional()
+        .setTitle('Children')
+        .setDescription('Array of component IDs that should be displayed in the summary'),
+    ).onlyIn(Variant.External),
+  )
+  .addProperty(new CG.prop('childComponents', new CG.arr(CG.layoutNode)).onlyIn(Variant.Internal))
+  .addProperty(
+    new CG.prop(
+      'children',
+      new CG.arr(new CG.str())
+        .optional()
+        .setTitle('Children')
+        .setDescription('Array of component IDs that should be displayed in the summary'),
+    ).onlyIn(Variant.External),
+  )
+  .addProperty(
+    new CG.prop(
+      'whatToRender',
+      new CG.obj(
+        new CG.prop(
+          'type',
+          new CG.enum('page', 'layoutSet', 'component', 'task')
+            .optional({ default: 'component' })
+            .setTitle('Mode')
+            .setDescription('Config for what should be rendered'),
+        ),
+        new CG.prop('id', new CG.str()),
+      ),
+    ),
+  )
+  .addProperty(
+    new CG.prop(
+      'overWriteProperties',
+      new CG.arr(
+        new CG.union(
+          INPUT_SUMMARY_PROPS,
+          CHECKBOX_SUMMARY_PROPS,
+          RADIO_SUMMARY_PROPS,
+          DROPDOWN_SUMMARY_PROPS,
+          MULTIPLE_SELECT_SUMMARY_PROPS,
+        ),
+      ).optional(),
+    ),
+  );
```

Comment:
Ahh, now I get why you have `componentId` in overrides. How about making this a map instead, where the key is the component ID? That would make for faster lookups and it would be impossible to mess things up by adding multiple overrides for the same component. Also, nitfy having a union for all of these - but does it work? I.e. do you only get possible options in vscode for an Input component when the type has narrowed down to that type?